src/accelerate/utils/dataclasses.py

```diff
@@ -1312,3 +1320,132 @@ def set_tensorboard_logging_options(self):
+
+
+@dataclass
+class BnbQuantizationConfig:
+    """
+    A plugin to enable BitsAndBytes 4bit and 8bit quantization
+    """
+
+    load_in_8bit: bool = field(default=False, metadata={"help": "enable 8bit quantization."})
+
+    llm_int8_threshold: float = field(default=6.0, metadata={"help": "value of the outliner threshold."})
+
+    load_in_4bit: bool = field(default=False, metadata={"help": "enable 4bit quantization."})
+
+    bnb_4bit_quant_type: str = field(
+        default="fp4",
+        metadata={
+            "help": "set the quantization data type in the `bnb.nn.Linear4Bit` layers. Options are {'fp4','np4'}."
+        },
+    )
+
+    bnb_4bit_use_double_quant: bool = field(
+        default=False,
+        metadata={
+            "help": "enable nested quantization where the quantization constants from the first quantization are quantized again."
+        },
+    )
+
+    bnb_4bit_compute_dtype: bool = field(
+        default="fp32",
+        metadata={
+            "help": "This sets the computational type which might be different than the input time. For example, inputs might be "
+            "fp32, but computation can be set to bf16 for speedups. Options are {'fp32','fp16','bf16'}."
+        },
+    )
+
+    torch_dtype: torch.dtype = field(
+        default=torch.float16,
+        metadata={
+            "help": "this sets the dtype of the remaining non quantized layers. `bitsandbytes` library suggests to set the value to `torch.float16`."
+        },
+    )
+
+    skip_modules: List[str] = field(
+        default=None,
+        metadata={"help": "an explicit list of the modules that we don't quantize. We keep them in `torch.dtype`."},
+    )
+
+    keep_in_fp32_modules: List[str] = field(
+        default=None,
+        metadata={"help": "an explicit list of the modules that we don't quantize. We keep them in `torch.float32`."},
+    )
+
+    # we will see if it will be useful
+    enable_fp32_cpu_offload: bool = field(
+        default=False,
+        metadata={
+            "help": """ this flag is used for advanced use cases and users that are aware of this feature. If you want to split
+            your model in different parts and run some parts in int8 on GPU and some parts in fp32 on CPU, you can use
+            this flag. This is useful for offloading large models such as `google/flan-t5-xxl`. Note that the int8
+            operations will not be run on CPU."""
+        },
+    )
+    target_dtype: Union[torch.dtype, CustomDtype] = field(
+        default=None, metadata={"help": "dtype of the quantized layers"}
+    )
+
+    def __post_init__(self):
+        """
+        Safety checker that arguments are correct - also replaces some NoneType arguments with their default values.
+        """
+        if not isinstance(self.load_in_8bit, bool):
+            raise ValueError("load_in_8bit must be a boolean")
+
+        if not isinstance(self.load_in_4bit, bool):
```

Comment:
Also I would check something else here, if load_in_4bit is activated and `llm_int8_threshold` is different than the default value raise an error or warning saying that `llm_int8_threshold` can be used only for `load_in_8bit` - this can be also added on transformers

Reply:
check added